x-pack/plugins/ml/public/application/routing/use_resolver.ts

```diff
@@ -5,115 +5,153 @@
  * 2.0.
  */
 
-import { useEffect, useState } from 'react';
-import type { IUiSettingsClient } from '@kbn/core/public';
+import { useCallback, useEffect, useRef, useState } from 'react';
 import { i18n } from '@kbn/i18n';
-import type { DataViewsContract } from '@kbn/data-views-plugin/public';
-import { getSavedSearch } from '../util/dependency_cache';
-import {
-  getDataViewById,
-  getDataViewAndSavedSearch,
-  type DataViewAndSavedSearch,
-} from '../util/index_utils';
-import { createSearchItems } from '../jobs/new_job/utils/new_job_utils';
+import useMount from 'react-use/lib/useMount';
+import { PLUGIN_ID } from '../../../common/constants/app';
+import { showExpiredLicenseWarning } from '../license/expired_warning';
+import { useMlKibana, useMlLicenseInfo } from '../contexts/kibana';
+import { type MlCapabilitiesKey } from '../../../common/types/capabilities';
+import { usePermissionCheck } from '../capabilities/check_capabilities';
 import type { ResolverResults, Resolvers } from './resolvers';
 import type { MlContextValue } from '../contexts/ml';
-import { useNotifications } from '../contexts/kibana';
-import { useCreateAndNavigateToMlLink } from '../contexts/kibana/use_create_url';
-import { ML_PAGES } from '../../../common/constants/locator';
+import { ML_APP_LOCATOR, ML_PAGES } from '../../../common/constants/locator';
 
 /**
- * Hook to resolve route specific requirements
- * @param dataViewId optional Kibana data view id, used for wizards
- * @param savedSearchId optional Kibana saved search id, used for wizards
- * @param config Kibana UI Settings
- * @param resolvers an array of resolvers to be executed for the route
- * @return { context, results } returns the ML context and resolver results
+ * Resolves required dependencies for landing on the page
+ * and performs redirects if needed.
+ *
+ * @param requiredLicence
+ * @param requiredCapabilities
  */
-export const useResolver = (
-  dataViewId: string | undefined,
-  savedSearchId: string | undefined,
-  config: IUiSettingsClient,
-  dataViewsContract: DataViewsContract,
-  resolvers: Resolvers
-): { context: MlContextValue; results: ResolverResults } => {
-  const notifications = useNotifications();
-
-  const funcNames = Object.keys(resolvers); // Object.entries gets this wrong?!
-  const funcs = Object.values(resolvers); // Object.entries gets this wrong?!
-  const tempResults = funcNames.reduce((p, c) => {
-    p[c] = {};
-    return p;
-  }, {} as ResolverResults);
-
-  const [context, setContext] = useState<any | null>(null);
-  const [results, setResults] = useState(tempResults);
-  const redirectToJobsManagementPage = useCreateAndNavigateToMlLink(
-    ML_PAGES.ANOMALY_DETECTION_JOBS_MANAGE
-  );
+export const useRouteResolver = (
+  requiredLicence: 'full' | 'basic',
```

Comment:
nit, we use the American spelling "license"

Reply:
renamed in https://github.com/elastic/kibana/pull/159176/commits/ae8e55edc00f8b16f4d135e32533b7b2729f8a3a